src/components/product.js

```diff
@@ -580,6 +610,40 @@ export default class Product extends Component {
     }
   }
 
+  onCarouselItemClick(evt, target) {
+    const selectedImageId = target.dataset.imageId;
+    const imageList = this.model.images;
+    const foundImage = imageList.find((image) => {
+      return image.id === parseInt(selectedImageId, 10);
+    });
+
+    if (foundImage) {
+      this.selectedImage = foundImage;
+    }
+    this.view.render();
+  }
+
+  onCarouselChange(toRight) {
+    const imageList = this.model.images;
+    const currentImage = imageList.find((image) => {
+      return image.id === this.currentImage.id;
+    });
+    const currentImageIndex = imageList.indexOf(currentImage);
+    if (toRight) {
```

Comment:
instead of passing through `isRight`, why not pass through `1` or `-1` ? might simplify things a bit. And maybe extract that logic into its own function so we can do early returns instead of `else`. 

```
 nextIndex(currentIndex, offset) {
  const nextIndex = currentIndex + offset;
  if (nextIndex > this.model.images.length) {
   return 0;
  }
  if (nextIndex < 0) {
    return this.model.images.length -1;
  }
  return nextIndex;
}
```